test/tests/functional/pbs_apbasil_inventory.py

```diff
@@ -224,3 +224,40 @@ def test_InventoryVnodes(self):
                 # Compare xml vnode with pbs node.
                 print "Validating vnode:%s" % (vnode['vnode'])
                 self.comp_node(vnode)
+
+    def test_cray_login_node_memory_unit(self):
+        """
+        This test validates that cray mom node resources value remain
+        unchanged before and after adding $alps_client in mom config.
+        """
+        mom_id = self.mom.shortname
+        try:
+            cray_login_node = self.server.status(NODE, id=mom_id)[0]
+        except PbsStatusError:
+            self.assertFalse(
+                "Cray login node %s doesn't exist on pbs server" % (mom_id))
+
+        self.mom.unset_mom_config('$alps_client', False)
+        self.reset_nodes(mom_id)
+
+        try:
+            pbs_node = self.server.status(NODE, id=mom_id)[0]
+
+        except PbsStatusError:
+            self.assertFalse(
+                "Cray login node %s doesn't exist on pbs server" % (mom_id))
+
```

Comment:
Hi Dilip,
Can't we have Line 235 and 244 on same try block?

Reply:
Hi @riyazhakki , 
No we can't have them in same try block, because the call in line 244 is after calling reset_nodes() function, which deletes all the existing node and add the mom node again.

Regards
Dilip

Reply:
I feel a try/except block can be avoided if we keep line 235,240,241 and 244 in one try block.
What say?

Reply:
Just as a note, the correct usage of assertFalse() is self.assertFalse(<condition>, msg).  With your usage, the message will be considered true, so it will fail, but will not print the message.  If you want it to always fail,  you want self.assertFalse(True, <msg>) (or alternately use assertTrue(False), <msg>).

Reply:
I agree with Riyaz.  You can merge these two try/except blocks by putting all the code into one (including the removal of alps_client and reset_nodes().  You won't be able to really tell which of the two failed, but I don't think it is really necessary.

Reply:
Hi @riyazhakki, @bhroam ,
    Differentiating which, out of the two node failed was the exact reason why two try blocks were needed. Since these nodes are results of two different set of code.One is created the default way of creating node, while cray login node is created through alps code. Also i noticed that I kept the log message same in both the places, which is a mistake.   

Unless you guys really want me to merge these try block, i would like to keep it with better log message.

Warm regards
Dilip

Reply:
Hey @dilip-krishnan, merging these two try block looks ok to me unless you feel necessary to print the distinct messages for each exception.